### PR TITLE
Add circleci yaml, output raw test output and xml report.

### PR DESCRIPTION
**Pull request by wallnerryan**
Add the circle CI yaml to add timezone and output `go test` output to a file downloadable from the UI.

alt text
alt text

This also outputs XML test output like https://33-43121312-gh.circle-artifacts.com/0//tmp/circle-junit.GMLBqU2/report.xml so that [test summaries](https://circleci.com/gh/ClusterHQ/flocker-go/33#tests) are produced

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+machine:
+  timezone:
+    America/Los_Angeles
+
+# Output the test output to circle.
+test:
+  override:
+    - go test -v ./... > test.out
```
Comment (wallrj) [inline circle.yml]: I noticed that the default command used by circle in e.g. https://circleci.com/gh/ClusterHQ/flocker-go/28#tests  is `go test -v -race ./...`

Comment (wallnerryan) [inline circle.yml]: thanks, will change that back

**Comment by wallrj** [comment]
Thanks @wallnerryan 

Looks good.

So is it the presence of the `test.out` artifact that causes the build result to appear in the github PR? 
Or is that a separate configuration change?
Because I don't see the `All checks have passed` message in other flocker-go PRs.

If not, then I guess my question is....what would I do with the test.out file?
Also does it cost us any extra to store these artifacts?

But please merge if you're happy with everything.

**Comment by wallnerryan** [comment]
Actually, i made a permissions change so we could see the circleci tests passing here in the PRs. Other PRs dont have them because they dont have any recent commits. My thought process is that its not so easy to see the test output like we do on travis if some build fails. So one would use the test output to debug. 

also, I may try to configure something like this (https://circleci.com/docs/test-metadata/#metadata-collection-in-custom-test-steps) instead which seems like it may be more intuitive.

Thanks for the review.

**Pull request merged**